mimic/model/heat_objects.py

```diff
@@ -9,6 +9,27 @@
 from mimic.model.behaviors import BehaviorRegistryCollection, EventDescription
 
 
+def get_stack_or_404(f):
+    """
+    Decorator which retrieves a stack from a collection and adds it to the
+    function's arguments if it exists. If the stack does not exist, a 404
```

Comment:
I don't really like how it changes the function's signature either. I reverted it because I have another idea of how to do it, but I'll have to try it later.